packages/x-tree-view/src/internals/plugins/useTreeViewFocus/useTreeViewFocus.ts

```diff
@@ -10,28 +10,20 @@ import { useInstanceEventHandler } from '../../hooks/useInstanceEventHandler';
 export const useTreeViewFocus: TreeViewPlugin<UseTreeViewFocusSignature> = ({
   instance,
   params,
-  state,
-  setState,
   models,
   rootRef,
 }) => {
-  const setFocusedNodeId = useEventCallback((nodeId: React.SetStateAction<string | null>) => {
-    const cleanNodeId = typeof nodeId === 'function' ? nodeId(state.focusedNodeId) : nodeId;
-    setState((prevState) => ({ ...prevState, focusedNodeId: cleanNodeId }));
-  });
-
   const isNodeFocused = React.useCallback(
-    (nodeId: string) => state.focusedNodeId === nodeId,
-    [state.focusedNodeId],
+    (nodeId: string) => models.focusedNodeId.value === nodeId,
+    [models.focusedNodeId.value],
   );
 
   const focusNode = useEventCallback((event: React.SyntheticEvent, nodeId: string | null) => {
     if (nodeId) {
-      setFocusedNodeId(nodeId);
-
       if (params.onNodeFocus) {
         params.onNodeFocus(event, nodeId);
```

Comment:
Should we rename this `onFocusedNodeIdChange` to be consistent with the other model?

And maybe we should call the model `focusedNode` instead of `focusedNodeId`, we went for `expandedNodes` and `selectedNodes` without the `Id` suffix.
In which case the callback would be `onFocusedNodeChange`